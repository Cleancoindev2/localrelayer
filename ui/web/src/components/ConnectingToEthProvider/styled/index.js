import styled from 'styled-components';

export const EthMissMainWrapper = styled.div`
  display: flex;
  flex-direction: column;
  min-height: 100vh;
  border: 2px solid 0F2335;
  border-radius: 5px;
  align-items: center;
  justify-content: center;
  font-size: 1.5rem;
  background-color: #0F2335;
`;

export const MessageWrapper = styled.div`
  display: flex;
  text-align: center;
  color: white;
`;

export const MetaMaskLink = styled.a.attrs({
  rel: 'noopener noreferrer',
  target: '_blank',
  href: 'https://metamask.io/',
})`
  margin-left: 5px;
  margin-right: 5px;
`;
