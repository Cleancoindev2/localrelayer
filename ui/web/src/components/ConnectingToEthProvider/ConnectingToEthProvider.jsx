import React from 'react';
import {
  Icon,
} from 'antd';

import * as S from './styled';


const ConnectingToEthProvider = () => (
  <S.EthMissMainWrapper>
    <S.MessageWrapper>
      <div>
        <Icon
          type="exclamation-circle"
          theme="filled"
        />
        You are not connected to Ethereum
        <div>
          Please
          <S.MetaMaskLink>
            download Metamask
          </S.MetaMaskLink>
          to use the exchange
        </div>
      </div>
    </S.MessageWrapper>
  </S.EthMissMainWrapper>

);

export default ConnectingToEthProvider;
