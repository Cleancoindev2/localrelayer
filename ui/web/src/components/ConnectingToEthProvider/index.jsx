import ConnectingToEthProvider from './ConnectingToEthProvider';

export default ConnectingToEthProvider;
