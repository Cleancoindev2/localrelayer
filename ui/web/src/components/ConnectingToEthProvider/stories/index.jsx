import {
  storiesOf,
} from '@storybook/react';
import ConnectingToEthProvider from '..';


storiesOf('Components|ConnectingToEthProvider', module)
  .add(
    'default',
    ConnectingToEthProvider,
  )
  .add(
    'full screen',
    ConnectingToEthProvider,
    {
      options: {
        goFullScreen: true,
      },
    },
  );
