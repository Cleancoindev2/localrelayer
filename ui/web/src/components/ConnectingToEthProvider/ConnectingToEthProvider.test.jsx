import React from 'react';
import renderer from 'react-test-renderer';

import ConnectingToEthProvider from './ConnectingToEthProvider';


test('ConnectingToEthProvider component', () => {
  expect(
    renderer.create(
      <ConnectingToEthProvider />,
    ).toJSON(),
  ).toMatchSnapshot();
});
