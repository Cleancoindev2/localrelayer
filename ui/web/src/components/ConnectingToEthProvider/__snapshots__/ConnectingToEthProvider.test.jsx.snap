// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectingToEthProvider component 1`] = `
<div
  className="styled__EthMissMainWrapper-sc-8po40b-0 jOoMWO"
>
  <div
    className="styled__MessageWrapper-sc-8po40b-1 kdyXGL"
  >
    <div>
      <i
        className="anticon anticon-exclamation-circle"
      >
        <svg
          aria-hidden="true"
          className=""
          data-icon="exclamation-circle"
          fill="currentColor"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm-32 232c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V296zm32 440a48.01 48.01 0 0 1 0-96 48.01 48.01 0 0 1 0 96z"
          />
        </svg>
      </i>
      You are not connected to Ethereum
      <div>
        Please
        <a
          className="styled__MetaMaskLink-sc-8po40b-2 iWVqpL"
          href="https://metamask.io/"
          rel="noopener noreferrer"
          target="_blank"
        >
          download Metamask
        </a>
        to use the exchange
      </div>
    </div>
  </div>
</div>
`;
